src/core/ClearStack.js

```diff
@@ -70,7 +70,9 @@ getJasmineRequireObj().clearStack = function(j$) {
 
     const SAFARI =
       global.navigator &&
-      /^((?!chrome|android).)*safari/i.test(global.navigator.userAgent);
+      /(^((?!chrome|android).)*safari)|(^((?!chrome|android|firefox).)+$)/i.test(
```

Comment:
If I'm reading it correctly, the added bit of regex says that every user agent string that doesn't include "chrome", "android", or "firefox" should get the Safari treatment. I think that's too broad, especially since the Safari mode is known to be slower in other browsers. I think the regex should specifically target the Playwright Webkit user agent string.

Additionally:

* This needs a unit test, like the other tests in ClearStackSpec.js. Otherwise it's very likely that your change will be broken by the next person who touches this code.
* The name of the SAFARI constants should be updated.

Reply:
I'll add a test and update the variable name. Further scoping the regex may not be possible, unfortunately, as the user agent string is almost identical to Chrome's, minus the "Chrome" part at the end (if I recall correctly). I can dig a little more and see if I can find another reliable way to distinguish it, though.

Reply:
I changed my added regex pattern so that it matches any user agent string that ends with `Win64; x64) AppleWebKit/[0-9.]+ (KHTML, like Gecko)`. Does that seem acceptably specific? As I mentioned before, the string I'm trying to match is pretty generic: `Mozilla/5.0 (Windows NT 6.2; Win64; x64) AppleWebKit/605.1.15 (KHTML, like Gecko)`.